### ISSUE TITLE
Implementation of Asymtotic Decider for MarchingCubes

### ISSUE THREAD

**Issue by KhoiDOO** [KhoiDOO/conquer3d#2]
- Compute the edge cases
- Implement Trilinear Interpolation scheme for resolving the edge cases in Marching Cubes

**Comment by KhoiDOO** [comment]
Presequisite for Dual Marching Cubes

**Comment by KhoiDOO** [comment]
### Implementation Completed in PR #8

The **Marching Cubes with Asymptotic Decider (MCA)** has been fully implemented in CUDA and exposed via `conquer3d.ops.marching_cubes_asymptotic` (alias `mca`).

#### Key Features:
- **Topologically Consistent**: Resolves all 6 ambiguous bilinear face cases dynamically on the GPU using Nielson-Hamann face saddles $S = \frac{f_0 f_2 - f_1 f_3}{(f_0 + f_2) - (f_1 + f_3)}$, guaranteeing watertight 2-manifold surfaces with 0 cracks.
- **Hybrid Fast-Path (`cubeFaceAmbigMask`)**: Over 85-90% of active voxels bypass face/loop tracing and evaluate directly in $\mathcal{O}(1)$ via constant lookup tables.
- **Differentiable**: Custom analytical backward gradient kernel (`backward_dmca_kernel`) w.r.t. scalar SDF values and vertex colors.
- **High Throughput**: Extracts 3.3M faces in **3.01 ms** on RTX 4090 ($1024^3$ grid).

#### Usage Example:
```python
import torch
from conquer3d.data_structure import create_voxel_grid
from conquer3d.ops import marching_cubes_asymptotic, mca

# 1. Create dense or sparse voxel grid
grid_vertices, voxels, _ = create_voxel_grid(
    grid_min=[-1.0, -1.0, -1.0],
    grid_max=[1.0, 1.0, 1.0],
    res=[128, 128, 128],
    device="cuda"
)

# 2. Evaluate scalar SDF values at grid vertices
sdf = torch.norm(grid_vertices, dim=-1) - 0.7

# 3. Extract watertight 2-manifold surface mesh
vertices, faces = marching_cubes_asymptotic(grid_vertices, voxels, sdf, iso=0.0)
print(f"Extracted watertight mesh: {vertices.shape[0]:,} vertices, {faces.shape[0]:,} faces")
```
